lib/recurly/hosted-field.js

```diff
@@ -95,27 +95,48 @@ export class HostedField extends Emitter {
   }
 
   inject () {
-    this.target.innerHTML = `
-      <div class="${this.classList}">
-        <iframe
-          src="${this.url}"
-          border="0"
-          frameborder="0"
-          allowtransparency="true"
-          scrolling="no">
-        </iframe>
-      </div>
-    `;
+    this.target.innerHTML = this.fieldHtml();
+    this.container = this.target.children[0]; // recurly-hosted-field
 
-    this.container = this.target.children[0];
-    this.iframe = this.container.children[0];
-    this.window = this.iframe.contentWindow;
+    if (isMobile()) {
+      this.injectMobileElements();
+    } else {
+      this.iframe = this.container.children[0];
+    }
 
+    this.window = this.iframe.contentWindow;
     this.iframe.style.height = '100%';
     this.iframe.style.width = '100%';
     this.iframe.style.background = 'transparent';
   }
 
+  injectMobileElements() {
+    this.tabbingProxy = this.container.children[0];
+    this.iframe = this.container.children[1];
+    this.bindDeferredFocusOnTabbingProxy();
+  }
+
+  fieldHtml () {
+    const isMobile = isMobile();
+    return `<div class="${this.classList}">
+      ${isMobile ? dom.createHiddenInput().outerHTML : ''}
+      <iframe
+        src="${this.url}"
+        border="0"
+        frameborder="0"
+        allowtransparency="true"
+        scrolling="no">
+      </iframe>
+      </div>
+      ${isMobile && this.config.singleField ? dom.createHiddenInput().outerHTML : ''}
```

Comment:
I don't think it's necessary for a singular hosted field needs any special treatment here. It should be treated the same as if there were multiple, no?

Reply:
A bit of a nitpick, but would you mind going back to the former nesting pattern in this html template? It's a bit easier to read.